In [1]:
import pandas as pd
import numpy as np
import copy
import re

In [2]:
import pandas as pd
test = pd.read_csv("amazon_reviews_us_Kitchen_v1_00.tsv", sep = '\t',error_bad_lines = False)
test['label'] = -1
a1 = test.loc[test['star_rating']==1].sample(50000)
a2 = test.loc[test['star_rating']==2].sample(50000)
a3 = test.loc[test['star_rating']==3].sample(50000)
a4 = test.loc[test['star_rating']==4].sample(50000)
a5 = test.loc[test['star_rating']==5].sample(50000)


b'Skipping line 16148: expected 15 fields, saw 22\nSkipping line 20100: expected 15 fields, saw 22\nSkipping line 45178: expected 15 fields, saw 22\nSkipping line 48700: expected 15 fields, saw 22\nSkipping line 63331: expected 15 fields, saw 22\n'
b'Skipping line 86053: expected 15 fields, saw 22\nSkipping line 88858: expected 15 fields, saw 22\nSkipping line 115017: expected 15 fields, saw 22\n'
b'Skipping line 137366: expected 15 fields, saw 22\nSkipping line 139110: expected 15 fields, saw 22\nSkipping line 165540: expected 15 fields, saw 22\nSkipping line 171813: expected 15 fields, saw 22\n'
b'Skipping line 203723: expected 15 fields, saw 22\nSkipping line 209366: expected 15 fields, saw 22\nSkipping line 211310: expected 15 fields, saw 22\nSkipping line 246351: expected 15 fields, saw 22\nSkipping line 252364: expected 15 fields, saw 22\n'
b'Skipping line 267003: expected 15 fields, saw 22\nSkipping line 268957: expected 15 fields, saw 22\nSkipping line 303336: expected 15 field

b'Skipping line 1968846: expected 15 fields, saw 22\nSkipping line 1999941: expected 15 fields, saw 22\nSkipping line 2001492: expected 15 fields, saw 22\nSkipping line 2011204: expected 15 fields, saw 22\nSkipping line 2025295: expected 15 fields, saw 22\n'
b'Skipping line 2041266: expected 15 fields, saw 22\nSkipping line 2073314: expected 15 fields, saw 22\nSkipping line 2080133: expected 15 fields, saw 22\nSkipping line 2088521: expected 15 fields, saw 22\n'
b'Skipping line 2103490: expected 15 fields, saw 22\nSkipping line 2115278: expected 15 fields, saw 22\nSkipping line 2153174: expected 15 fields, saw 22\nSkipping line 2161731: expected 15 fields, saw 22\n'
b'Skipping line 2165250: expected 15 fields, saw 22\nSkipping line 2175132: expected 15 fields, saw 22\nSkipping line 2206817: expected 15 fields, saw 22\nSkipping line 2215848: expected 15 fields, saw 22\nSkipping line 2223811: expected 15 fields, saw 22\n'
b'Skipping line 2257265: expected 15 fields, saw 22\nSkipping line

In [3]:
new_test = pd.concat([a1,a2,a3,a4,a5])

In [4]:
new_test['lable'] = 1

In [5]:
new_test.label[test.star_rating>3] = 1
new_test.label[test.star_rating<3] = 2
new_test.label[test.star_rating==3] = 3
new_test = new_test[['label','review_body']] 

/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so w

In [6]:
copy_test = copy.deepcopy(new_test)

In [7]:
import gensim
model_1 = gensim.models.Word2Vec.load('save')

In [8]:
#data clearing
copy_test['review_body'] = copy_test['review_body'].str.lower()

def tag(x):
    return re.sub('<.*?>','',str(x))
copy_test['review_body'] = copy_test['review_body'].apply(lambda x:tag(x))

def url(x):
    return re.sub('(https?|ftp|file)://[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]','',str(x))

copy_test['review_body'] = copy_test['review_body'].apply(lambda x:url(x))

import contractions
copy_test['review_body'] = copy_test['review_body'].apply(lambda x:contractions.fix(x))

def non_alphabetical(x):
    return re.sub('[^a-zA-Z\s]','',str(x))
copy_test['review_body'] = copy_test['review_body'].apply(lambda x:non_alphabetical(x))
def extra_space(x):
    return re.sub( ' +',' ',str(x))
copy_test['review_body'] = copy_test['review_body'].apply(lambda x:extra_space(x))

In [9]:
#Pre-processing

import nltk
from nltk.corpus import stopwords
nltk.download('stopwords')
stop_words_set = set(stopwords.words('english'))
from nltk import word_tokenize, pos_tag

def stop_words(x):
    word_tokens = word_tokenize(x)
    temp = []
    for i in word_tokens:
        if i not in stop_words_set:
            temp.append(i)
    return temp
copy_test['review_body'] = copy_test['review_body'].apply(lambda x:stop_words(x))

[nltk_data] Downloading package stopwords to
[nltk_data]     /Users/chenlin/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


In [10]:
ternary_copy = copy_test

In [11]:
ternary_copy

label                                        review_body
2901789      2  [griddle, works, finebut, plates, rust, cost, ...
4079589      2  [badthe, blade, sharp, enough, cut, tomato, fi...
461277       2  [products, like, actually, make, productionmar...
2583042      2  [second, time, washed, largest, container, dis...
3856377      2  [admit, love, keurig, coffee, brewer, said, as...
...        ...                                                ...
2397865      1  [oven, pantry, delivered, stated, specificatio...
519273       1                                             [like]
4557793      1  [ratings, already, given, product, reason, pur...
428144       1  [purchased, gold, igrind, could, happier, come...
963579       1                                             [huge]

[250000 rows x 2 columns]

In [12]:
def vector_avg(x):
    i = 0
    n = 0
    a = np.zeros(300)
    while i<len(x):
        if x[i] in model_1.wv:
            a += model_1.wv[x[i]]
        else:
            n += 1
        i = i+1
    if n == 0: return a
    else:return a/(len(x)-n)

In [13]:
copy_test['avg_vector'] = copy_test['review_body'].apply(lambda x:vector_avg(x))

In [14]:
binary_copy = copy_test.loc[copy_test['label']!=3]

# binary_copy

In [15]:
binary_copy['y'] = 1

/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [16]:
binary_copy.y[binary_copy['label']==2] = 0

/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/opt/anaconda3/lib/python3.7/site-packages/pandas/core/series.py:992: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._where(~key, value, inplace=True)


In [17]:
binary_copy

label                                        review_body  \
2901789      2  [griddle, works, finebut, plates, rust, cost, ...   
4079589      2  [badthe, blade, sharp, enough, cut, tomato, fi...   
461277       2  [products, like, actually, make, productionmar...   
2583042      2  [second, time, washed, largest, container, dis...   
3856377      2  [admit, love, keurig, coffee, brewer, said, as...   
...        ...                                                ...   
2397865      1  [oven, pantry, delivered, stated, specificatio...   
519273       1                                             [like]   
4557793      1  [ratings, already, given, product, reason, pur...   
428144       1  [purchased, gold, igrind, could, happier, come...   
963579       1                                             [huge]   

                                                avg_vector  y  
2901789  [0.024641906699308984, 0.08881746848615316, 0....  0  
4079589  [-0.016323742822610905, 0.05006689679742392, -...  0  
461277   [0.06468200084054843, -0.010757067662780173, 0...  0  
2583042  [0.05246899649500847, 1.5992026403546333, 0.13...  0  
3856377  [0.06659146030120719, 0.03285078117067912, 0.0...  0  
...                                                    ... ..  
2397865  [0.8622930869460106, 1.1786859966814518, -0.08...  1  
519273   [0.04800622910261154, 0.02565152198076248, 0.1...  1  
4557793  [0.025994234447382757, 0.05975001984367483, -0...  1  
428144   [0.017926203673899483, 0.167255596448538, -0.0...  1  
963579   [-0.0970420092344284, 0.042717333883047104, 0....  1  

[200000 rows x 4 columns]

In [18]:
X = binary_copy['avg_vector']
Y = binary_copy['y']

In [19]:
from sklearn.model_selection import train_test_split

In [20]:
x_train, x_test, y_train, y_test = train_test_split(X, Y, random_state = 19, test_size = 0.2)

In [21]:
x_train = np.array(x_train.tolist())

y_train = np.array(y_train.tolist())

x_train = np.nan_to_num(x_train)

In [22]:
x_test = np.array(x_test.tolist())

y_test = np.array(y_test.tolist())

x_test = np.nan_to_num(x_test)

In [23]:
# import libraries
import torch
from torch.utils.data import DataLoader, Dataset
import torchvision
import torchvision.transforms as transforms
from torch.utils.data.sampler import SubsetRandomSampler
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

In [24]:
from torch.utils.data import DataLoader, Dataset
from torch.utils.data import TensorDataset, DataLoader

In [25]:

# number of subprocesses to use for data loading
num_workers = 0
# how many samples per batch to load
batch_size = 20
# percentage of training set to use as validation
#valid_size = 0.2

DatasetTrain = TensorDataset(torch.from_numpy(x_train),torch.from_numpy(y_train))
        
DatasetTest=TensorDataset(torch.from_numpy(x_test),torch.from_numpy(y_test))

train_loader=torch.utils.data.DataLoader(DatasetTrain,batch_size=batch_size,shuffle=True,drop_last=True, num_workers=0)

valid_loader=torch.utils.data.DataLoader(DatasetTest, batch_size=batch_size, drop_last=True,num_workers=0)

In [26]:
import torch.nn as nn
import torch.nn.functional as F

# define the NN architecture
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        # number of hidden nodes in each layer 
        hidden_1 = 50
        hidden_2 = 10
        # linear layer (300 -> hidden_1)
        self.fc1 = nn.Linear(300, hidden_1)
        # linear layer (n_hidden -> hidden_2)
        self.fc2 = nn.Linear(hidden_1, hidden_2)
        # linear layer (n_hidden -> 10)
        self.fc3 = nn.Linear(hidden_2, 2)
        # dropout layer (p=0.2)
        # dropout prevents overfitting of data
        self.dropout = nn.Dropout(0.2)

    def forward(self, x):
        # flatten image input
        x = x.view(-1, 300)
        # add hidden layer, with relu activation function
        x = F.relu(self.fc1(x))
        # add dropout layer
        x = self.dropout(x)
        # add hidden layer, with relu activation function
        x = F.relu(self.fc2(x))
        # add dropout layer
        x = self.dropout(x)
        # add output layer
        x = self.fc3(x)
        return x

# initialize the NN
model = Net()
print(model)

Net(
  (fc1): Linear(in_features=300, out_features=50, bias=True)
  (fc2): Linear(in_features=50, out_features=10, bias=True)
  (fc3): Linear(in_features=10, out_features=2, bias=True)
  (dropout): Dropout(p=0.2, inplace=False)
)


In [27]:
# import libraries
import torch
## Specify loss and optimization functions

# specify loss function
criterion = nn.CrossEntropyLoss()

# specify optimizer
optimizer = torch.optim.SGD(model.parameters(), lr=0.01)

In [28]:
train_loader

In [29]:
# number of epochs to train the model
n_epochs = 20  # suggest training between 20-50 epochs

model.train() # prep model for training

for epoch in range(n_epochs):
    # monitor training loss
    train_loss = 0.0
    
    ###################
    # train the model #
    ###################
    for data, target in train_loader:
        # clear the gradients of all optimized variables
        optimizer.zero_grad()
        # forward pass: compute predicted outputs by passing inputs to the model
        output = model(data.float())
        # calculate the loss
        loss = criterion(output, target)
        # backward pass: compute gradient of the loss with respect to model parameters
        loss.backward()
        # perform a single optimization step (parameter update)
        optimizer.step()
        # update running training loss
        train_loss += loss.item()*data.size(0)
        
    # print training statistics 
    # calculate average loss over an epoch
    train_loss = train_loss/len(train_loader.dataset)

    print('Epoch: {} \tTraining Loss: {:.6f}'.format(
        epoch+1, 
        train_loss
        ))

Epoch: 1 	Training Loss: 0.483913
Epoch: 2 	Training Loss: 0.392670
Epoch: 3 	Training Loss: 0.378242
Epoch: 4 	Training Loss: 0.370848
Epoch: 5 	Training Loss: 0.365118
Epoch: 6 	Training Loss: 0.361285
Epoch: 7 	Training Loss: 0.356872
Epoch: 8 	Training Loss: 0.353547
Epoch: 9 	Training Loss: 0.351196
Epoch: 10 	Training Loss: 0.348159
Epoch: 11 	Training Loss: 0.347070
Epoch: 12 	Training Loss: 0.343552
Epoch: 13 	Training Loss: 0.342804
Epoch: 14 	Training Loss: 0.341586
Epoch: 15 	Training Loss: 0.338453
Epoch: 16 	Training Loss: 0.339810
Epoch: 17 	Training Loss: 0.337566
Epoch: 18 	Training Loss: 0.336264
Epoch: 19 	Training Loss: 0.335728
Epoch: 20 	Training Loss: 0.334147


In [30]:
model

Net(
  (fc1): Linear(in_features=300, out_features=50, bias=True)
  (fc2): Linear(in_features=50, out_features=10, bias=True)
  (fc3): Linear(in_features=10, out_features=2, bias=True)
  (dropout): Dropout(p=0.2, inplace=False)
)

In [31]:
DatasetTest=TensorDataset(torch.from_numpy(x_test),torch.from_numpy(y_test))

total_num = 0
correct = 0
for x_temp,y_real in DatasetTest:
    res = model(x_temp.float())
    _, pred = torch.max(res,1)
    total_num += 1
    if int(pred) == int(y_real):
        correct += 1

In [32]:
if 0 in copy_test['label'].values:
    print('hhhh')

In [33]:
accuracy_mymodel_avg =  correct/total_num

In [34]:
accuracy_mymodel_avg

0.854175

In [35]:
import copy

In [36]:
import gensim
model_1 = gensim.models.Word2Vec.load('save')

In [37]:
def vector_concat(x):
    i = 0
    n_x = 0
    length = len(x)
    if length == 0:
        return np.zeros(3000)
    a = np.zeros(300)
    
    while i<10:
        if n_x < length:
            if x[n_x] in model_1.wv :
                if i == 0:
                    ans = copy.deepcopy(model_1.wv[x[n_x]])
                    i += 1
                else:
                    ans = np.concatenate((ans,model_1.wv[x[n_x]]), axis=None)
                    i += 1
            n_x += 1
        else:
            if i == 0:
                ans = a
                i += 1
            else:
                ans = np.concatenate((ans,a), axis=None)
                i += 1
    return ans

In [38]:
binary_copy['concat_vec'] = binary_copy['review_body'].apply(lambda x:vector_concat(x))

/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [39]:
binary_copy

label                                        review_body  \
2901789      2  [griddle, works, finebut, plates, rust, cost, ...   
4079589      2  [badthe, blade, sharp, enough, cut, tomato, fi...   
461277       2  [products, like, actually, make, productionmar...   
2583042      2  [second, time, washed, largest, container, dis...   
3856377      2  [admit, love, keurig, coffee, brewer, said, as...   
...        ...                                                ...   
2397865      1  [oven, pantry, delivered, stated, specificatio...   
519273       1                                             [like]   
4557793      1  [ratings, already, given, product, reason, pur...   
428144       1  [purchased, gold, igrind, could, happier, come...   
963579       1                                             [huge]   

                                                avg_vector  y  \
2901789  [0.024641906699308984, 0.08881746848615316, 0....  0   
4079589  [-0.016323742822610905, 0.05006689679742392, -...  0   
461277   [0.06468200084054843, -0.010757067662780173, 0...  0   
2583042  [0.05246899649500847, 1.5992026403546333, 0.13...  0   
3856377  [0.06659146030120719, 0.03285078117067912, 0.0...  0   
...                                                    ... ..   
2397865  [0.8622930869460106, 1.1786859966814518, -0.08...  1   
519273   [0.04800622910261154, 0.02565152198076248, 0.1...  1   
4557793  [0.025994234447382757, 0.05975001984367483, -0...  1   
428144   [0.017926203673899483, 0.167255596448538, -0.0...  1   
963579   [-0.0970420092344284, 0.042717333883047104, 0....  1   

                                                concat_vec  
2901789  [-0.04459453, -0.07535303, -0.4262394, -0.0587...  
4079589  [-0.020585256, 0.03667236, -0.26981658, 0.1782...  
461277   [-0.013184801, -0.005327605, 0.17744249, 0.035...  
2583042  [-0.069223836, 0.11872141, 0.029483773, 0.0597...  
3856377  [-0.10600603, 0.029664135, -0.09290071, 0.0523...  
...                                                    ...  
2397865  [0.07905123, 0.053250413, -0.01902701, 0.03232...  
519273   [0.04800622910261154, 0.02565152198076248, 0.1...  
4557793  [0.22660527, -0.20551513, -0.020619687, 0.1293...  
428144   [0.07608983, -0.057236966, 0.3409337, -0.27152...  
963579   [-0.0970420092344284, 0.042717333883047104, 0....  

[200000 rows x 5 columns]

In [40]:
X = binary_copy['concat_vec']
Y = binary_copy['y']

In [41]:
x_train, x_test, y_train, y_test = train_test_split(X, Y, random_state = 19, test_size = 0.2)

In [42]:
x_train = np.array(x_train.tolist())

y_train = np.array(y_train.tolist())

x_train = np.nan_to_num(x_train)

In [43]:
x_test = np.array(x_test.tolist())

y_test = np.array(y_test.tolist())

x_test = np.nan_to_num(x_test)

In [44]:

# number of subprocesses to use for data loading
num_workers = 0
# how many samples per batch to load
batch_size = 20
# percentage of training set to use as validation
#valid_size = 0.2

DatasetTrain = TensorDataset(torch.from_numpy(x_train),torch.from_numpy(y_train))
        
DatasetTest=TensorDataset(torch.from_numpy(x_test),torch.from_numpy(y_test))

train_loader=torch.utils.data.DataLoader(DatasetTrain,batch_size=batch_size,shuffle=True,drop_last=True, num_workers=0)

valid_loader=torch.utils.data.DataLoader(DatasetTest, batch_size=batch_size, drop_last=True,num_workers=0)

In [45]:
y_train

array([0, 1, 1, ..., 1, 0, 0])

In [46]:
import torch.nn as nn
import torch.nn.functional as F

# define the NN architecture
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        # number of hidden nodes in each layer 
        hidden_1 = 50
        hidden_2 = 10
        # linear layer (300 -> hidden_1)
        self.fc1 = nn.Linear(3000, hidden_1)
        # linear layer (n_hidden -> hidden_2)
        self.fc2 = nn.Linear(hidden_1, hidden_2)
        # linear layer (n_hidden -> 10)
        self.fc3 = nn.Linear(hidden_2, 2)
        # dropout layer (p=0.2)
        # dropout prevents overfitting of data
        self.dropout = nn.Dropout(0.2)

    def forward(self, x):
        # flatten image input
        x = x.view(-1, 3000)
        # add hidden layer, with relu activation function
        x = F.relu(self.fc1(x))
        # add dropout layer
        x = self.dropout(x)
        # add hidden layer, with relu activation function
        x = F.relu(self.fc2(x))
        # add dropout layer
        x = self.dropout(x)
        # add output layer
        x = self.fc3(x)
        return x

# initialize the NN
model = Net()

In [47]:
print(model)

Net(
  (fc1): Linear(in_features=3000, out_features=50, bias=True)
  (fc2): Linear(in_features=50, out_features=10, bias=True)
  (fc3): Linear(in_features=10, out_features=2, bias=True)
  (dropout): Dropout(p=0.2, inplace=False)
)


In [48]:
# specify loss function
criterion = nn.CrossEntropyLoss()

# specify optimizer
optimizer = torch.optim.SGD(model.parameters(), lr=0.01)

In [49]:
# number of epochs to train the model
n_epochs = 20  # suggest training between 20-50 epochs

model.train() # prep model for training

for epoch in range(n_epochs):
    # monitor training loss
    train_loss = 0.0
    
    ###################
    # train the model #
    ###################
    for data, target in train_loader:
        # clear the gradients of all optimized variables
        optimizer.zero_grad()
        # forward pass: compute predicted outputs by passing inputs to the model
        output = model(data.float())
        # calculate the loss
        loss = criterion(output, target)
        # backward pass: compute gradient of the loss with respect to model parameters
        loss.backward()
        # perform a single optimization step (parameter update)
        optimizer.step()
        # update running training loss
        train_loss += loss.item()*data.size(0)
        
    # print training statistics 
    # calculate average loss over an epoch
    train_loss = train_loss/len(train_loader.dataset)

    print('Epoch: {} \tTraining Loss: {:.6f}'.format(
        epoch+1, 
        train_loss
        ))

Epoch: 1 	Training Loss: 0.569580
Epoch: 2 	Training Loss: 0.464485
Epoch: 3 	Training Loss: 0.440153
Epoch: 4 	Training Loss: 0.424263
Epoch: 5 	Training Loss: 0.409070
Epoch: 6 	Training Loss: 0.393365
Epoch: 7 	Training Loss: 0.378543
Epoch: 8 	Training Loss: 0.363366
Epoch: 9 	Training Loss: 0.346821
Epoch: 10 	Training Loss: 0.331147
Epoch: 11 	Training Loss: 0.317480
Epoch: 12 	Training Loss: 0.302490
Epoch: 13 	Training Loss: 0.288299
Epoch: 14 	Training Loss: 0.275590
Epoch: 15 	Training Loss: 0.265235
Epoch: 16 	Training Loss: 0.254121
Epoch: 17 	Training Loss: 0.243234
Epoch: 18 	Training Loss: 0.233545
Epoch: 19 	Training Loss: 0.224100
Epoch: 20 	Training Loss: 0.217472


In [50]:
DatasetTest=TensorDataset(torch.from_numpy(x_test),torch.from_numpy(y_test))

total_num = 0
correct = 0
for x_temp,y_real in DatasetTest:
    res = model(x_temp.float())
    _, pred = torch.max(res,1)
    total_num += 1
    if int(pred) == int(y_real):
        correct += 1

In [51]:
accuracy_mymodel_concat =  correct/total_num

In [54]:
accuracy_mymodel_concat

0.77565

In [52]:
accuracy_mymodel_avg

0.854175

# conclusion:
the accuracy for mymodel using avg vector with binary label is 0.854175

the accuracy for mymodel using concat vector with binary label is 0.77565
